modules/gdscript/editor/script_templates/CharacterBody3D/godot_character_controller.gd

```diff
@@ -0,0 +1,73 @@
+# meta-description: Godot character controller for gravity games (FPS, TPS, ...)
+
+extends _BASE_
+
+
+@export_group("Kinematic")
+@export var walk_max_acceleration := 300.0
+@export var walk_acceleration := 15.0
+@export var run_max_acceleration := walk_max_acceleration * 1.5
+@export var run_acceleration := walk_acceleration * 1.5
+@export var friction := 200.0
+@export var jump_velocity := 5.0
+@export_group("Camera")
+@export var mouse_sensitivity := 0.005
+@export var camera_max_vertical_angle := 30
+@export var camera_min_vertical_angle := -10
+
+# Get the global gravity from the project settings. RigidBody3D nodes also use this value.
+var gravity: float = ProjectSettings.get_setting("physics/3d/default_gravity")
+var _current_acceleration := 0.0
+
+# Nodes for camera management, replace these nodes by yours and remove the initialization in _on_ready.
+var spring_arm := SpringArm3D.new()
+var camera := Camera3D.new()
+
+func _unhandled_input(event):
+	# The mouse cursor is not visible in the game, you can toogle the behavior with the escape key.
+	if event.is_action_pressed("ui_cancel"):
+		Input.set_mouse_mode(Input.MOUSE_MODE_CAPTURED if Input.get_mouse_mode() != Input.MOUSE_MODE_CAPTURED else Input.MOUSE_MODE_VISIBLE)
+
+	# Moves the camera according to the movement of the cursor.
+	elif Input.get_mouse_mode() == Input.MOUSE_MODE_CAPTURED and event is InputEventMouseMotion:
+		rotate_y(-event.relative.x * mouse_sensitivity)
+		spring_arm.rotate_x(-event.relative.y * mouse_sensitivity)
+		spring_arm.rotation.x = clamp(spring_arm.rotation.x, deg_to_rad(camera_min_vertical_angle), deg_to_rad(camera_max_vertical_angle))
```

Comment:
See my previous comment here when the templates were initially added https://github.com/godotengine/godot/pull/53957#pullrequestreview-816910090

I think that camera movement should be a separate script that you attach to a child SpringArm node for the player's head. That way the camera movement is self-contained in a separate script. Bonus: We could make a separate version for Camera3D so that we have a character movement script, an FPS camera script, and a TPS camera script. For left/right rotation, it could either rotate the parent character body, or it could rotate itself, we could pick one or maybe support both.